### PR TITLE
Add ServerBeanType unit test

### PR DESCRIPTION
**Pull request by odockal**
Signed-off-by: Ondrej Dockal <odockal@redhat.com>

### CODE REVIEW

**Comment by odockal** [comment]
@robstryker Would please take a look? It is quite hard to get familiar with core implementation and so I used existing mocking classes and changed a bit some implementation...Thanks

File: runtimes/tests/org.jboss.tools.rsp.server.wildfly.test/src/main/java/org/jboss/tools/rsp/server/wildfly/test/beans/MockServerCreationUtilities.java [hunk 1]
```diff
@@ -550,8 +583,13 @@ private static void createTmpDir() {
 	protected static File extractToTmpDir(String path) {
 		File pathFile = new File(MOCK_TMP_DIR, path);
 		if( !pathFile.exists()) {
-			ClassLoader classLoader = MockServerCreationUtilities.class.getClassLoader();
```
Comment (robstryker) [inline runtimes/tests/org.jboss.tools.rsp.server.wildfly.test/src/main/java/org/jboss/tools/rsp/server/wildfly/test/beans/MockServerCreationUtilities.java]: Any reasons you're switching this? I'm not convinced that's the right answer...

**Comment by robstryker** [comment]
Looks good to me, builds locally, can't see any obvious errors. I'll push it

File: runtimes/tests/org.jboss.tools.rsp.server.wildfly.test/src/main/java/org/jboss/tools/rsp/server/wildfly/test/beans/MockServerCreationUtilities.java (hunk 1 above)
Comment (odockal) [reply runtimes/tests/org.jboss.tools.rsp.server.wildfly.test/src/main/java/org/jboss/tools/rsp/server/wildfly/test/beans/MockServerCreationUtilities.java]: When running test from eclipse, the class is not on the class path and it does not find server mock folder... That was the reason, I believe that there could be better way, though... This came to me like straightforward and easy solution working for both cases (running from ide and from maven).

**Pull request merged**